packages/libs/react-ui/src/components/Form/TextField/TextField.tsx

```diff
@@ -1,35 +1,162 @@
+  );
+
+  const { hoverProps, isHovered } = useHover(props);
+  const { isFocused, isFocusVisible, focusProps } = useFocusRing({
+    isTextInput: true,
+    autoFocus: props.autoFocus,
+  });
+
+  // handle uncontrollable form state eg. react-hook-form
+  const handleOnChange = useCallback(
+    (event: ChangeEvent<ElementRef<'input'>>) => {
+      inputProps.onChange?.(event);
+      props.onChange?.(event);
+    },
+    [props.onChange, inputProps.onChange],
+  );
+
+  // aggregate error message from validation props
+  const errorMessage =
+    typeof props.errorMessage === 'function'
+      ? props.errorMessage(validation)
+      : props.errorMessage ?? validation.validationErrors.join(' ');
 
   return (
-    <div className={statusVal ? statusVariant[statusVal] : undefined}>
-      {label !== undefined && (
-        <FormFieldHeader htmlFor={id} label={label} tag={tag} info={info} />
+    <div className={classNames(formField, props.className)}>
+      {props.label && (
+        <FormFieldHeader
+          label={props.label}
+          tag={props.tag}
+          info={props.info}
+          {...labelProps}
+        />
       )}
-      <Input ref={ref} disabled={disabled} id={id} {...inputProps} />
-      {Boolean(helperText) && status !== 'negative' && (
-        <FormFieldHelper>{helperText}</FormFieldHelper>
+      <div className={inputContainer}>
+        {props.startAddon && (
+          <div
+            className={startAddon}
+            ref={(el) => {
+              if (el) {
+                ref.current?.style.setProperty(
+                  '--start-addon-width',
+                  `${el.offsetWidth}px`,
+                );
+              }
+            }}
+          >
+            {props.startAddon}
+          </div>
+        )}
+
+        <input
+          {...mergeProps(inputProps, focusProps, hoverProps)}
+          onChange={handleOnChange}
+          ref={ref}
+          className={input}
+          data-focused={isFocused || undefined}
+          data-disabled={props.disabled || undefined}
+          data-hovered={isHovered || undefined}
+          data-focus-visible={isFocusVisible || undefined}
+          data-invalid={validation.isInvalid || undefined}
+          data-positive={props.isPositive || undefined}
+          data-has-start-addon={!!props.startAddon || undefined}
+          data-has-end-addon={!!props.endAddon || undefined}
+        />
+
+        {props.endAddon && (
+          <div
+            className={endAddon}
+            ref={(el) => {
+              if (el) {
+                ref.current?.style.setProperty(
+                  '--end-addon-width',
+                  `${el.offsetWidth}px`,
+                );
+              }
+            }}
+          >
+            {props.endAddon}
+          </div>
+        )}
+      </div>
+
+      {props.description && !validation.isInvalid && (
+        <FormFieldHelper
+          {...descriptionProps}
+          intent={props.isPositive ? 'positive' : 'info'}
```

Comment:
@isaozler Does this mean we will no longer have a warning style?

Reply:
do we even need the positive prop?

Reply:
it is in the design so I added it, but would be happy to not have it :D